tools/RAiDER/models/gmao.py

```diff
@@ -0,0 +1,171 @@
+        DT = self._time - T0
+        time_ind = int(DT.total_seconds() / 3600.0 / 3.0)
+        
+        ml_min = 0
+        ml_max = 71
+
+        # open the dataset and pull the data
+        import pydap.client
+        import pydap.cas.urs
+        url = 'https://opendap.nccs.nasa.gov/dods/GEOS-5/fp/0.25_deg/assim/inst3_3d_asm_Nv'
+        session = pydap.cas.urs.setup_session('username', 'password', check_url=url)
+        ds = pydap.client.open_url(url, session=session)
+        
+        q = ds['qv'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        p = ds['pl'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        t = ds['t'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        h = ds['h'].array[time_ind,ml_min:(ml_max+1),lat_min_ind:(lat_max_ind+1),lon_min_ind:(lon_max_ind+1)][0]
+        
+        # calculate the lat, lon and mean h for each layer in the regular grid
+        hs = np.zeros(72)
+        for ii in range(72):
+            hs[ii] = np.mean(h[ii,:,:])
+        
+        lats = np.arange((-90 + lat_min_ind * self._lat_res), (-90 + (lat_max_ind+1) * self._lat_res), self._lat_res)
+        lons = np.arange((-180 + lon_min_ind * self._lon_res), (-180 + (lon_max_ind+1) * self._lon_res), self._lon_res)
+        
+        # restructure the 3-D lat/lon/h in regular grid
+        _lons = np.broadcast_to(lons[np.newaxis, np.newaxis, :],t.shape)
+        _lats = np.broadcast_to(lats[np.newaxis, :, np.newaxis],t.shape)
+        _hs = np.broadcast_to(hs[:, np.newaxis, np.newaxis],t.shape)
+        
+        # Re-structure everything from (heights, lats, lons) to (lons, lats, heights)
+        p = np.transpose(p)
+        q = np.transpose(q)
+        t = np.transpose(t)
+        h = np.transpose(h)
+        _lats = np.transpose(_lats)
+        _lons = np.transpose(_lons)
+        _hs = np.transpose(_hs)
+        
+        
+        # check this
+        # data cube format should be lats,lons,heights
+        p = p.swapaxes(0,1)
+        q = q.swapaxes(0,1)
+        t = t.swapaxes(0,1)
+        h = h.swapaxes(0,1)
+        _lats = _lats.swapaxes(0,1)
+        _lons = _lons.swapaxes(0,1)
+        _hs = _hs.swapaxes(0,1)
+        
+        
+        # For some reason z is opposite the others
+        p = np.flip(p, axis = 2)
+        q = np.flip(q, axis = 2)
+        t = np.flip(t, axis = 2)
+        h = np.flip(h, axis = 2)
+        _lats = np.flip(_lats, axis = 2)
+        _lons = np.flip(_lons, axis = 2)
+        _hs = np.flip(_hs, axis = 2)
+        
+
+        # interpolate (p,q,t) along the vertical axis from irregular grid to regular grid
+        from RAiDER.interpolator import interp_along_axis
+        p_intrpl = interp_along_axis(h, _hs, p, axis = 2)
+        q_intrpl = interp_along_axis(h, _hs, q, axis = 2)
+        t_intrpl = interp_along_axis(h, _hs, t, axis = 2)
```

Comment:
This stuff looks very similar to `_uniform_in_z`. Is there some way we can refactor it, instead of having duplicate implementations?

Reply:
@Askaholic @jlmaurer i assume this should be recurrent in other models too for model levels?